### ISSUE TITLE
Ingest Content for Content Provider - UBFRANKFURT

### ISSUE THREAD

**Issue by lobajuluwa** [gbhl/bhl-europe#328]
Task description: Align (UBFRANKFURT) upload data/structure with ingest tool needs

subtask: Ingest (UBFRANKFURT) data

**Comment by chris-sleep** [comment]
- Folder Names ok (one so far, one volume in a series)
- Folder Structure: ok (serial on article level structure)
- Folder Naming
  
  @wkollernhm can you please take a look at UBFrankfurt/BJ2248281/BJ2248281_0001_2248284 and confirm if you can see any issue with the naming? The article subfolders don't fully match FSG but I'm not sure if this will be an issue for the PI tool.
- File Naming
  - InternalIdentifier ok
  - FileSequenceNumber ok
  - PageType ok
  - PrintedPageNumber ok

**Comment by chris-sleep** [comment]
11 metadata files in 11 folders; mods

**Comment by melitabirthaelmer** [comment]
Information for mapping:

Acts as aggregator.
Will use MODS element "mods:recordContentSource" for indicating additional data provider.

**Comment by wkollernhm** [comment]
Just to clarify: this feature is not yet implemented but recorded and on the todo list - see issue #365

**Comment by melitabirthaelmer** [comment]
CP asked for feedback. Do we have an answer for the naming of the subfolders? Can we test it with the PI tool?
@chris-sleep

**Comment by wkollernhm** [comment]
The example upload looks fine. @chris-sleep where exactly did you see problems with the article sub-folders?

**Comment by chris-sleep** [comment]
@wkollernhm not problems per se; I wanted to be certain my interpretation of the serial on article layout was not missing anything - this was the first I'd assessed.

**Comment by chris-sleep** [comment]
running a test PI on selected content: de-ubfrankfurt/BJ2248281/BJ2248281_0001_2248284

metadata is mods so using SMT config: -m c -cm 6 -if <input_file> -of <output_file>

Final .aip preparation complete - errors thrown: 

Error: Filename convention broken! rename your page files according to File Submission Guidelines (FSG). (can the offending filename be attached to the error perhaps?)

@ZhengLIAtos - can you please take a look at the generated .aip 
@wkollernhm - can you let me know if the generated OLEF is good for this one? (I'm wondering if the FSG errors reported are because the aip data is not viable, or if those are warnings and the aip is viable but the ingested package will be missing functionality)

**Comment by wkollernhm** [comment]
The error message indicates, that some filename couldn't be parsed. This means data in the OLEF will be missing (like page-type and page-number). I've created an issue for this (#383).

**Comment by melitabirthaelmer** [comment]
CP will provide metadata as from now via OAI-PMH. 
Metadata available as DC, MODS or METS. Structured data is only delivered by METS.

Already uploaded content (BJ2248281) corresponds to the OAI identifier except the "BJ" infront.

Please retest with OAI metadata.

**Comment by wkollernhm** [comment]
Folder names must EXACTLY equal the OAI-identifier. Therefor the folder name is invalid. However we can mass-rename the content on our end, we just should communicate this to the CP.

**Comment by chris-sleep** [comment]
@wkollernhm on that basis I've copied the only upload so far to be:

de-ubfrankfurt/oai:sammlungen.ub.uni-frankfurt.de:2248281/ - is this correct?  (I've not so far renamed anything below that point)

-- update --

if I set the web service to be http://[.......................]/oai?verb=GetRecord&metadataPrefix=mods&identifier=oai:sammlungen.ub.uni-frankfurt.de:

then I can use the folder name simply de-ubfrankfurt/test-2248281/2248281.

The web service presents 10 records, all of which have the same identifier prefix, do you think this is a viable approach?

**Comment by chris-sleep** [comment]
(cannot yet test full PI until serial content support in place)

**Comment by wkollernhm** [comment]
Yes, the OAI-PMH setup should work that way, if the prefix stays static. Only problem I see so far is the metadata for the volume / article level information. Since those folders should conform to the same naming scheme as well and return a valid metadata entry from OAI-PMH. Will that be the case? (is volume / article metadata available via OAi-PMH at all?)

**Comment by wkollernhm** [comment]
I've done a test ingest with the data recently provided. Besides some minor fixes (content missed an empty section-level folder) the content seems to work fine.
Sent feedback to UB-Frankfurt and waiting for ok to proceed to productive.
